html/mycakeapp/src/Controller/AuctionController.php

```diff
@@ -84,19 +88,48 @@ public function add()
 	{
 		// Biditemインスタンスを用意
 		$biditem = $this->Biditems->newEntity();
-		// POST送信時の処理
-		if ($this->request->is('post')) {
-			// $biditemにフォームの送信内容を反映
-			$biditem = $this->Biditems->patchEntity($biditem, $this->request->getData());
-			// $biditemを保存する
-			if ($this->Biditems->save($biditem)) {
-				// 成功時のメッセージ
-				$this->Flash->success(__('保存しました。'));
-				// トップページ（index）に移動
-				return $this->redirect(['action' => 'index']);
+
+		$file = $this->request->getData('image_path');
+		$imagepath_3 = substr($file['name'], -3);
+		$imagepath_4 = substr($file['name'], -4);
+		$image_last_3 = ($imagepath_3 === 'jpg' || $imagepath_3 === 'JPG' || $imagepath_3 === 'png' || $imagepath_3 === 'PNG' || $imagepath_3 === 'gif' ||  $imagepath_3 === 'GIF');
+		$image_last_4 = ($imagepath_4 === 'jpeg' || $imagepath_4 === 'JPEG');
```

Comment:
この比較ではだめというわけではないですが、長いのでこうしましょう！
```suggestion
		$image_last_3 = in_array($imagepath_3, ['jpg', 'JPG', 'png', 'PNG', 'gif', 'GIF',]);
		$image_last_4 = in_array($imagepath_4, ['jpeg', 'JPEG',]);
```